### ISSUE TITLE
bug: unsafe violation on ValueRef

### ISSUE THREAD

**Issue by andylokandy** [wangrunji0408/flat-json#6]
https://github.com/risingwavelabs/jsonbb/blob/bc93534cc38a11ca01da5fbe32eb53d09350e54b/src/value_ref.rs#L207-L210

This unsafe is not always valid because user can construct the ValueRef via `ValueRef::String("hello world")`

**Comment by wangrunji0408** [comment]
Yes. A possible solution is to wrap the &str in a new type to prevent user from constructing it.

**Comment by BugenZhao** [comment]
+1. This can be easily triggered by `array_push` as it called `as_slice` inside.

```rust
let mut array: Value = "[1]".parse().unwrap();
array.array_push(ValueRef::String("str"));
println!("{}", array);
```

I'm considering whether we should require `ValueRef` to be created from a `Value`, ensuring it cannot exist independently.
